### PR TITLE
malachite_bigint

### PR DESCRIPTION
**Pull request by youknowone**
test of https://github.com/RustPython/Parser/pull/18

### CODE REVIEW

**Comment by youknowone** [comment]
@MichaReiser looking good?

File: Cargo.lock [hunk 1]
```diff
@@ -197,6 +197,22 @@ version = "2.3.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "6776fc96284a0bb647b615056fc496d1fe1644a7ab01829818a6d91cae888b84"
 
+[[package]]
+name = "block-buffer"
```
Comment (MichaReiser) [inline Cargo.lock:201]: Hmm. This is adding a lot new dependencies.

**Review by MichaReiser — COMMENTED** [review]
We need to upgrade Ruff to the latest RustPython version first to measure the bigint change in isolation (e.g does the diff include the statement parsing changes?)

**Comment by MichaReiser** [comment]
I ran the benchmarks locally and can't see a significant change in performance. This doesn't mean that performance isn't improving, we just don't see any improvements for our benchmarks. For example, it could just be that our sample files use too few ints for it to be significant. 

<pre>group                                      main                                   malachite
-----                                      ----                                   ---------
linter/all-rules/large/dataset.py          1.01      9.0±0.04ms     4.5 MB/sec<span style="color:#26A269"><b>    1.00      8.9±0.19ms     4.6 MB/sec</b></span>
linter/all-rules/numpy/ctypeslib.py        1.01      2.1±0.02ms     8.0 MB/sec<span style="color:#26A269"><b>    1.00      2.1±0.03ms     8.1 MB/sec</b></span>
linter/all-rules/numpy/globals.py          1.01    226.1±0.82µs    13.1 MB/sec<span style="color:#26A269"><b>    1.00    224.0±0.47µs    13.2 MB/sec</b></span>
linter/all-rules/pydantic/types.py<span style="color:#26A269"><b>         1.00      3.6±0.02ms     7.1 MB/sec</b></span>    1.00      3.6±0.01ms     7.1 MB/sec
linter/default-rules/large/dataset.py      1.00      4.5±0.01ms     9.1 MB/sec<span style="color:#26A269"><b>    1.00      4.5±0.01ms     9.1 MB/sec</b></span>
linter/default-rules/numpy/ctypeslib.py    1.01    924.8±2.10µs    18.0 MB/sec<span style="color:#26A269"><b>    1.00    913.7±4.23µs    18.2 MB/sec</b></span>
linter/default-rules/numpy/globals.py<span style="color:#26A269"><b>      1.00     98.3±0.49µs    30.0 MB/sec</b></span>    1.01     99.1±0.25µs    29.8 MB/sec
linter/default-rules/pydantic/types.py<span style="color:#26A269"><b>     1.00   1930.8±4.73µs    13.2 MB/sec</b></span>    1.01  1951.5±13.11µs    13.1 MB/sec
parser/large/dataset.py<span style="color:#26A269"><b>                    1.00      3.4±0.09ms    12.0 MB/sec</b></span>    1.01      3.4±0.01ms    11.8 MB/sec
parser/numpy/ctypeslib.py                  1.00    664.9±2.10µs    25.0 MB/sec<span style="color:#26A269"><b>    1.00    661.9±1.81µs    25.2 MB/sec</b></span>
parser/numpy/globals.py<span style="color:#26A269"><b>                    1.00     64.3±1.24µs    45.9 MB/sec</b></span>    1.03     65.9±2.33µs    44.7 MB/sec
parser/pydantic/types.py                   1.01   1437.8±4.56µs    17.7 MB/sec<span style="color:#26A269"><b>    1.00  1424.7±12.89µs    17.9 MB/sec</b></span>
</pre>

I'm going to close this PR for now. You can run the benchmarks locally with

```
# Run baseline (once)
cargo benchmark --save-baseline main

# Make changes
cargo benchmark --save-baseline malachite

# compare baseline
critcmp main malachite
```

**Comment by youknowone** [comment]
The malachite patch may not affect Ruff that much. The underlying data structure is very similar. Operation is faster, but I don't think Ruff has int operation that much.

**Comment by youknowone** [comment]
Oh, yes, that's also right.

In future, I expect parser returns `&str` as unparsed int string and actual BigInt build can be deferred when it actually needed to be created. Does it make sense?

**Comment by MichaReiser** [comment]
That makes sense. That would also simplify the lexer

**Pull request closed without merging**